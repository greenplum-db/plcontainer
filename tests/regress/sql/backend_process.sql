\connect

SET plcontainer.backend_type = 'process';

CREATE OR REPLACE FUNCTION check_postgres() RETURNS bool AS $$
# container: plc_python_shared
import subprocess
try:
    output = subprocess.check_output(["ps", "-ef"], text=True, stderr=subprocess.STDOUT)
    return "postgres" in output
except subprocess.CalledProcessError as e:
    raise Exception(e.stdout)
$$ LANGUAGE plcontainer;

\! LOCAL_PROCESS_MODE=1 $GPHOME/bin/plcontainer_clients/py3client &> /tmp/plcontainer_py3client.log &

SELECT check_postgres();

SHOW plcontainer.backend_type;

\connect
