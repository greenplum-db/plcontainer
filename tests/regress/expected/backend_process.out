SET plcontainer.backend_type = 'process';
CREATE OR REPLACE FUNCTION hello() RETURNS text AS $$
# container: plc_python_shared
return 'hello'
$$ LANGUAGE plcontainer;
SELECT hello();
 hello 
-------
 hello
(1 row)

SHOW plcontainer.backend_type;
 plcontainer.backend_type 
--------------------------
 process
(1 row)

RESET plcontainer.backend_type;
SHOW plcontainer.backend_type;
 plcontainer.backend_type 
--------------------------
 docker
(1 row)

